src/Microsoft.DotNet.TestFramework/Microsoft.DotNet.TestFramework.TestInstance.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+
+namespace Microsoft.DotNet.TestFramework
+{
+    public class TestInstance
+    {
+        private string _testDestination;
+        private TestScenario _testScenario;
+
+        internal TestInstance(TestScenario testScenario, string testDestination)
+        {
+            if (testScenario == null)
+            {
+                throw new ArgumentNullException("testScenario");
+            }
+
+            if (string.IsNullOrEmpty(testDestination))
+            {
+                throw new ArgumentException("testDestination");
+            }
+
+            _testScenario = testScenario;
+            _testDestination = testDestination;
+
+            if (Directory.Exists(testDestination))
+            {
+                Directory.Delete(testDestination, true);
+            }
+
+            Directory.CreateDirectory(testDestination);
+            CopySource();
+        }
+
+        private void CopySource()
+        {
+            var sourceDirs = Directory.GetDirectories(_testScenario.SourceRoot, "*", SearchOption.AllDirectories)
+                                 .Where(dir =>
+                                 {
+                                     dir = dir.ToLower();
+                                     return !dir.EndsWith("\\bin") && !dir.Contains("\\bin\\")
```

Comment:
Does that work on other OSes? Maybe you wanted to use Path.Directory separator char?